plugin/pkg/scheduler/scheduler.go

```diff
@@ -236,6 +233,9 @@ func (sched *Scheduler) scheduleOne() {
 			})
 			return
 		}
+		if err := sched.config.SchedulerCache.FinishBinding(&assumed); err != nil {
```

Comment:
I am not sure why we shouldn't call FinishBinding when Bind fails. FinishBinding [marks the fact that binding has finished](https://github.com/kubernetes/kubernetes/blob/master/plugin/pkg/scheduler/schedulercache/cache.go#L148) in the state of the pod. This is [used later to clean up the pod](https://github.com/kubernetes/kubernetes/blob/master/plugin/pkg/scheduler/schedulercache/cache.go#L365). I think we should clean up these pods even if their binding has failed.